client/config.go

```diff
@@ -29,9 +29,9 @@ func (config Config) isSSL() bool {
 		return true
 	} else if configURL.Scheme == "http" {
 		return false
-	} else {
-		return false
 	}
+
+	return false
```

Comment:
:-1: To me, this is just harder to read broken out like this, particularly because this `return` is only for very unexpected cases (non-http URLs), and not part of the normal control flow... Maybe

```go
if configURL, err := ... {
   ...
} else {
    return configURL.Scheme == "https"
}
```

But I know, that still causes lint to complain about `if block ends with a return statement`...

Reply:
There are quite a few ways this could be written. I was also wondering if we should be handling / logging an error at this point if there was an issue parsing the URL?

We could of course revert back and keep the `return false` as the failing condition. Or maybe even use a switch rather than `else if else if`?

```go
configURL, err := url.Parse(config.URL)
if err != nil {
	// Maybe log that there has been an error parsing the URL here?
	return false
}

switch configURL.Scheme {
case "https":
	return true
case "http":
	return false
default:
	return false
}
```

I feel this separates the call to `url.Parse` from checking the contents of `configURL.Scheme`. What are your thoughts on this?